src/sources/audiosource.h

```diff
@@ -3,235 +3,197 @@
 
 #include "sources/urlresource.h"
 #include "util/audiosignal.h"
+#include "util/indexrange.h"
 #include "util/memory.h"
 #include "util/result.h"
 #include "util/samplebuffer.h"
 
+
 namespace mixxx {
 
 // forward declaration(s)
+class AudioSource;
 class AudioSourceConfig;
 
+typedef std::shared_ptr<AudioSource> AudioSourcePointer;
+
+// AudioSource v2 interface
+class IAudioSource {
+  public:
+    virtual ~IAudioSource() = default;
+
+    // Read sample frames from the requested frame index range. Only
+    // forward-oriented index ranges need to be supported. The implementation
+    // is responsible for restricting the requested range to the range that
+    // is actually readable.
+    //
+    // Only that part of the output buffer corresponding to the returned
+    // range is allowed to be modified. All remaining samples in the output
+    // buffer should stay untouched. The samples in the output buffer need
```

Comment:
That's the responsibility of the caller. We don't want to make any assumptions what the caller is doing with the results. Moreover we would need to add extra code to **each** of the implementations to fill the unread parts with zero (that maybe are never used by the caller anyway).